core/pipeline/label.go

```diff
@@ -97,6 +100,78 @@ func LoadVCSLabels(workdir string) []Label {
 	return labels
 }
 
+// Define a type for functions that fetch a branch commit
+type fetchFunc func(repo *git.Repository, branch string) (*object.Commit, error)
+
+// Function to fetch from the origin remote
+func fetchFromOrigin(repo *git.Repository, branch string) (*object.Commit, error) {
+	// Fetch from the origin remote
+	cmd := exec.Command("git", "fetch", "--depth", "1", "origin", branch)
+	err := cmd.Run()
+	if err != nil {
+		return nil, fmt.Errorf("error fetching branch from origin: %w", err)
+	}
+
+	// Get the reference of the fetched branch
+	refName := plumbing.ReferenceName(fmt.Sprintf("refs/remotes/origin/%s", branch))
+	ref, err := repo.Reference(refName, true)
+	if err != nil {
+		return nil, fmt.Errorf("error getting reference: %w", err)
+	}
+
+	// Get the commit object of the fetched branch
+	branchCommit, err := repo.CommitObject(ref.Hash())
+	if err != nil {
+		return nil, fmt.Errorf("error getting commit: %w", err)
+	}
+
+	return branchCommit, nil
+}
+
+// Function to fetch from the fork remote
+// GitHub forks are not added as remotes by default, so we need to guess the fork URL
+// This is a heuristic approach, as the fork might not exist from the information we have
+func fetchFromFork(repo *git.Repository, branch string) (*object.Commit, error) {
+	// Get the username of the person who initiated the workflow run
+	username := os.Getenv("GITHUB_ACTOR")
+
+	// Get the repository name (owner/repo)
+	repository := os.Getenv("GITHUB_REPOSITORY")
+	parts := strings.Split(repository, "/")
+	if len(parts) < 2 {
+		return nil, fmt.Errorf("invalid repository format: %s", repository)
+	}
+
+	forkURL := "https://github.com/" + username + "/" + parts[1]
```

Comment:
Agh, I don't love this, it feels very potentially hacky (one note, we can also use `GITHUB_SERVER_URL` here, to work with private GitHub instances).

That said, I don't have better suggestions if GitHub isn't doing anything to properly label their merge commits as auto-generated. I'd be happy to get this in sooner rather than later if this resolves 90% of cases (which it seems it probably will) - but think we'll have to revisit at some point since it's not just GitHub who does this behavior.

Reply:
Great comment, implemented